test/Parse/type_expr.swift

```diff
@@ -507,33 +464,12 @@ func functionType() {
   >.matches(
     (
       (
-        (P1) & Optional<Foo>.P2,
-        Gen<Foo>.Bar,
-        (Foo, [Int : Foo?].Type)
-      ) -> (
-        [Foo.Bar]
-      ) -> Array<Foo>.Element
-    ).self
-  )
-
-  // FIXME: Teach Sema to recognize this type expression.
-  CheckType<
-    (
-      P1 & P2,
-      Gen<Foo>.Bar,
-      (Foo, [Int : Foo?].Type)
-    ) -> (
-      [Foo.Bar]
-    ) -> Foo
-  >.matches(
-    (
-      ( // expected-error {{expected type before '->'}}
```

Comment:
What’s the reasons you deleted this test case?

Reply:
It was a duplicate test case without sugared type qualifiers. I left a FIXME to remove it once we can make the other one succeed.